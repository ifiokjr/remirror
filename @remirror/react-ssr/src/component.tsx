import { EditorState, ExtensionManager, PlainObject } from '@remirror/core';
import { mapProps, ReactSerializer } from '@remirror/renderer-react';
import React, { FC } from 'react';

export interface RemirrorSSRProps {
  state: EditorState;
  attributes: PlainObject;
  manager: ExtensionManager;
}

export const RemirrorSSR: FC<RemirrorSSRProps> = ({ attributes, manager, state }) => {
  const outerProps = mapProps(attributes);
  return (
    <div {...outerProps}>
      {ReactSerializer.fromExtensionManager(manager).serializeFragment(state.doc.content)}
    </div>
  );
};
