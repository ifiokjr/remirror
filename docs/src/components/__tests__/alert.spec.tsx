import { render } from '@test-utils';
import React from 'react';
import { Alert } from '../alert.component';

test('docz setup', () => {
  const { asFragment, rerender } = render(<Alert kind='info' />);
  expect(asFragment()).toMatchSnapshot();
  rerender(<Alert kind='negative' />);
  expect(asFragment()).toMatchSnapshot();
});
