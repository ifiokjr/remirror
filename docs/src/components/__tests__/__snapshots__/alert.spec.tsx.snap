// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docz setup 1`] = `
<DocumentFragment>
  <div
    class="css-10t5zsk-AlertStyled eto1ytx0"
    kind="info"
  />
</DocumentFragment>
`;

exports[`docz setup 2`] = `
<DocumentFragment>
  <div
    class="css-1wbxlgv-AlertStyled eto1ytx0"
    kind="negative"
  />
</DocumentFragment>
`;
